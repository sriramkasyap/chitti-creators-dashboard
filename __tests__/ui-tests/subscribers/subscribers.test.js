import { render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

import SubscriberList from "../../../src/components/SubscribersList/SubscribersList";

const subscribers = [
  {
    email: "sara.andersen@example.com",
    name: "Ms. Sara Andersen",
    subscriptionType: "Free",
  },
  {
    email: "edita.vestering@example.com",
    name: "Miss. Edita Vestering",
    subscriptionType: "Free",
  },
  {
    email: "roberto.vega@example.com",
    name: "Mr. Roberto Vega",
    subscriptionType: "Paid",
  },
  {
    email: "rudi.droste@example.com",
    name: "Mr. Rudi Droste",
    subscriptionType: "Free",
  },
  {
    email: "carolina.lima@example.com",
    name: "Mrs. Carolina Lima",
    subscriptionType: "Paid",
  },
  {
    email: "emre.asikoglu@example.com",
    name: "Mr. Emre Asikoglu",
    subscriptionType: "Free",
  },
  {
    email: "kent.brewer@example.com",
    name: "Mr. Kent Brewer",
    subscriptionType: "Free",
  },
  {
    email: "evan.carlson@example.com",
    name: "Mr. Evan Carlson",
    subscriptionType: "Free",
  },
  {
    email: "friedrich-karl.brand@example.com",
    name: "Mr. Friedrich-Karl Brand",
    subscriptionType: "Paid",
  },
  {
    email: "valentin.ortega@example.com",
    name: "Mr. Valentin Ortega",
    subscriptionType: "Free",
  },
];

const pagination = {
  limit: 10,
  page: 0,
};

const totalCount = 44;

let container;

beforeEach(() => {
  const mockSetPagination = jest.fn();
  container = render(
    <SubscriberList
      subscribers={subscribers}
      isLoading={false}
      error=""
      totalCount={totalCount}
      pagination={pagination}
      setPagination={mockSetPagination}
    />
  );
});

describe("Test Subscriber initial render", () => {
  it("should match the snapshot", () => {
    expect(container).toMatchSnapshot();
  });
});

describe("Test if no data received", () => {
  it("should render no data text if subscribers is empty", () => {
    const mockSetPagination = jest.fn();

    const { rerender, getByTestId } = container;

    rerender(
      <SubscriberList
        subscribers={[]}
        isLoading={false}
        error=""
        totalCount={0}
        pagination={pagination}
        setPagination={mockSetPagination}
      />
    );

    expect(getByTestId("no-subscribers-data")).toBeInTheDocument();
  });

  it("should render components on isLoading and error prop change", () => {
    const mockSetPagination = jest.fn();

    const { rerender, getByTestId } = container;

    rerender(
      <SubscriberList
        subscribers={subscribers}
        isLoading
        error=""
        totalCount={totalCount}
        pagination={pagination}
        setPagination={mockSetPagination}
      />
    );

    expect(getByTestId("loading-image")).toBeInTheDocument();

    rerender(
      <SubscriberList
        subscribers={subscribers}
        isLoading={false}
        error="Some error"
        totalCount={totalCount}
        pagination={pagination}
        setPagination={mockSetPagination}
      />
    );

    expect(getByTestId("notification")).toBeInTheDocument();
  });
});
