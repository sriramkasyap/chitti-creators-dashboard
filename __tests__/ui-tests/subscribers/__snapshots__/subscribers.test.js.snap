// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Subscriber initial render should match the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div
      id="chakra-toast-portal"
    >
      <ul
        id="chakra-toast-manager-top"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        id="chakra-toast-manager-top-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-top-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-bottom-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-bottom"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        id="chakra-toast-manager-bottom-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
    </div>
    <div>
      <div
        class="css-h53f3m"
      >
        <div
          class="css-13d8atv"
        >
          <table
            class="chakra-table css-0"
            role="table"
          >
            <thead
              class="css-0"
            >
              <tr
                class="css-0"
                role="row"
              >
                <th
                  class="css-mvn64f"
                  colspan="1"
                  role="columnheader"
                  style="cursor: pointer;"
                  title="Toggle SortBy"
                >
                  <div
                    class="css-u4p24i"
                  >
                    <p
                      class="chakra-text css-0"
                    >
                      Name
                    </p>
                    <span
                      class="css-rkk3mp"
                    />
                  </div>
                </th>
                <th
                  class="css-mvn64f"
                  colspan="1"
                  role="columnheader"
                  style="cursor: pointer;"
                  title="Toggle SortBy"
                >
                  <div
                    class="css-u4p24i"
                  >
                    <p
                      class="chakra-text css-0"
                    >
                      Email
                    </p>
                    <span
                      class="css-rkk3mp"
                    />
                  </div>
                </th>
                <th
                  class="css-mvn64f"
                  colspan="1"
                  role="columnheader"
                  style="cursor: pointer;"
                  title="Toggle SortBy"
                >
                  <div
                    class="css-u4p24i"
                  >
                    <p
                      class="chakra-text css-0"
                    >
                      Subscription Type
                    </p>
                    <span
                      class="css-rkk3mp"
                    />
                  </div>
                </th>
              </tr>
            </thead>
            <tbody
              class="css-0"
              role="rowgroup"
            >
              <tr
                class="css-0"
                role="row"
              >
                <td
                  class="css-j251j9"
                  role="cell"
                >
                  Ms. Sara Andersen
                </td>
                <td
                  class="css-j251j9"
                  role="cell"
                >
                  sara.andersen@example.com
                </td>
                <td
                  class="css-j251j9"
                  role="cell"
                >
                  Free
                </td>
              </tr>
              <tr
                class="css-0"
                role="row"
              >
                <td
                  class="css-j251j9"
                  role="cell"
                >
                  Miss. Edita Vestering
                </td>
                <td
                  class="css-j251j9"
                  role="cell"
                >
                  edita.vestering@example.com
                </td>
                <td
                  class="css-j251j9"
                  role="cell"
                >
                  Free
                </td>
              </tr>
              <tr
                class="css-0"
                role="row"
              >
                <td
                  class="css-j251j9"
                  role="cell"
                >
                  Mr. Roberto Vega
                </td>
                <td
                  class="css-j251j9"
                  role="cell"
                >
                  roberto.vega@example.com
                </td>
                <td
                  class="css-j251j9"
                  role="cell"
                >
                  Paid
                </td>
              </tr>
              <tr
                class="css-0"
                role="row"
              >
                <td
                  class="css-j251j9"
                  role="cell"
                >
                  Mr. Rudi Droste
                </td>
                <td
                  class="css-j251j9"
                  role="cell"
                >
                  rudi.droste@example.com
                </td>
                <td
                  class="css-j251j9"
                  role="cell"
                >
                  Free
                </td>
              </tr>
              <tr
                class="css-0"
                role="row"
              >
                <td
                  class="css-j251j9"
                  role="cell"
                >
                  Mrs. Carolina Lima
                </td>
                <td
                  class="css-j251j9"
                  role="cell"
                >
                  carolina.lima@example.com
                </td>
                <td
                  class="css-j251j9"
                  role="cell"
                >
                  Paid
                </td>
              </tr>
              <tr
                class="css-0"
                role="row"
              >
                <td
                  class="css-j251j9"
                  role="cell"
                >
                  Mr. Emre Asikoglu
                </td>
                <td
                  class="css-j251j9"
                  role="cell"
                >
                  emre.asikoglu@example.com
                </td>
                <td
                  class="css-j251j9"
                  role="cell"
                >
                  Free
                </td>
              </tr>
              <tr
                class="css-0"
                role="row"
              >
                <td
                  class="css-j251j9"
                  role="cell"
                >
                  Mr. Kent Brewer
                </td>
                <td
                  class="css-j251j9"
                  role="cell"
                >
                  kent.brewer@example.com
                </td>
                <td
                  class="css-j251j9"
                  role="cell"
                >
                  Free
                </td>
              </tr>
              <tr
                class="css-0"
                role="row"
              >
                <td
                  class="css-j251j9"
                  role="cell"
                >
                  Mr. Evan Carlson
                </td>
                <td
                  class="css-j251j9"
                  role="cell"
                >
                  evan.carlson@example.com
                </td>
                <td
                  class="css-j251j9"
                  role="cell"
                >
                  Free
                </td>
              </tr>
              <tr
                class="css-0"
                role="row"
              >
                <td
                  class="css-j251j9"
                  role="cell"
                >
                  Mr. Friedrich-Karl Brand
                </td>
                <td
                  class="css-j251j9"
                  role="cell"
                >
                  friedrich-karl.brand@example.com
                </td>
                <td
                  class="css-j251j9"
                  role="cell"
                >
                  Paid
                </td>
              </tr>
              <tr
                class="css-0"
                role="row"
              >
                <td
                  class="css-j251j9"
                  role="cell"
                >
                  Mr. Valentin Ortega
                </td>
                <td
                  class="css-j251j9"
                  role="cell"
                >
                  valentin.ortega@example.com
                </td>
                <td
                  class="css-j251j9"
                  role="cell"
                >
                  Free
                </td>
              </tr>
            </tbody>
          </table>
          <div
            class="css-3mgjd9"
          >
            <button
              class="chakra-button css-1y29bcw"
              data-testid="prev-btn"
              disabled=""
              type="button"
            >
              &lt; Previous
            </button>
            <p
              class="chakra-text css-0"
            >
              Page 
              1
               of 
              5
            </p>
            <button
              class="chakra-button css-1y29bcw"
              data-testid="next-btn"
              type="button"
            >
              Next &gt;
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-h53f3m"
    >
      <div
        class="css-13d8atv"
      >
        <table
          class="chakra-table css-0"
          role="table"
        >
          <thead
            class="css-0"
          >
            <tr
              class="css-0"
              role="row"
            >
              <th
                class="css-mvn64f"
                colspan="1"
                role="columnheader"
                style="cursor: pointer;"
                title="Toggle SortBy"
              >
                <div
                  class="css-u4p24i"
                >
                  <p
                    class="chakra-text css-0"
                  >
                    Name
                  </p>
                  <span
                    class="css-rkk3mp"
                  />
                </div>
              </th>
              <th
                class="css-mvn64f"
                colspan="1"
                role="columnheader"
                style="cursor: pointer;"
                title="Toggle SortBy"
              >
                <div
                  class="css-u4p24i"
                >
                  <p
                    class="chakra-text css-0"
                  >
                    Email
                  </p>
                  <span
                    class="css-rkk3mp"
                  />
                </div>
              </th>
              <th
                class="css-mvn64f"
                colspan="1"
                role="columnheader"
                style="cursor: pointer;"
                title="Toggle SortBy"
              >
                <div
                  class="css-u4p24i"
                >
                  <p
                    class="chakra-text css-0"
                  >
                    Subscription Type
                  </p>
                  <span
                    class="css-rkk3mp"
                  />
                </div>
              </th>
            </tr>
          </thead>
          <tbody
            class="css-0"
            role="rowgroup"
          >
            <tr
              class="css-0"
              role="row"
            >
              <td
                class="css-j251j9"
                role="cell"
              >
                Ms. Sara Andersen
              </td>
              <td
                class="css-j251j9"
                role="cell"
              >
                sara.andersen@example.com
              </td>
              <td
                class="css-j251j9"
                role="cell"
              >
                Free
              </td>
            </tr>
            <tr
              class="css-0"
              role="row"
            >
              <td
                class="css-j251j9"
                role="cell"
              >
                Miss. Edita Vestering
              </td>
              <td
                class="css-j251j9"
                role="cell"
              >
                edita.vestering@example.com
              </td>
              <td
                class="css-j251j9"
                role="cell"
              >
                Free
              </td>
            </tr>
            <tr
              class="css-0"
              role="row"
            >
              <td
                class="css-j251j9"
                role="cell"
              >
                Mr. Roberto Vega
              </td>
              <td
                class="css-j251j9"
                role="cell"
              >
                roberto.vega@example.com
              </td>
              <td
                class="css-j251j9"
                role="cell"
              >
                Paid
              </td>
            </tr>
            <tr
              class="css-0"
              role="row"
            >
              <td
                class="css-j251j9"
                role="cell"
              >
                Mr. Rudi Droste
              </td>
              <td
                class="css-j251j9"
                role="cell"
              >
                rudi.droste@example.com
              </td>
              <td
                class="css-j251j9"
                role="cell"
              >
                Free
              </td>
            </tr>
            <tr
              class="css-0"
              role="row"
            >
              <td
                class="css-j251j9"
                role="cell"
              >
                Mrs. Carolina Lima
              </td>
              <td
                class="css-j251j9"
                role="cell"
              >
                carolina.lima@example.com
              </td>
              <td
                class="css-j251j9"
                role="cell"
              >
                Paid
              </td>
            </tr>
            <tr
              class="css-0"
              role="row"
            >
              <td
                class="css-j251j9"
                role="cell"
              >
                Mr. Emre Asikoglu
              </td>
              <td
                class="css-j251j9"
                role="cell"
              >
                emre.asikoglu@example.com
              </td>
              <td
                class="css-j251j9"
                role="cell"
              >
                Free
              </td>
            </tr>
            <tr
              class="css-0"
              role="row"
            >
              <td
                class="css-j251j9"
                role="cell"
              >
                Mr. Kent Brewer
              </td>
              <td
                class="css-j251j9"
                role="cell"
              >
                kent.brewer@example.com
              </td>
              <td
                class="css-j251j9"
                role="cell"
              >
                Free
              </td>
            </tr>
            <tr
              class="css-0"
              role="row"
            >
              <td
                class="css-j251j9"
                role="cell"
              >
                Mr. Evan Carlson
              </td>
              <td
                class="css-j251j9"
                role="cell"
              >
                evan.carlson@example.com
              </td>
              <td
                class="css-j251j9"
                role="cell"
              >
                Free
              </td>
            </tr>
            <tr
              class="css-0"
              role="row"
            >
              <td
                class="css-j251j9"
                role="cell"
              >
                Mr. Friedrich-Karl Brand
              </td>
              <td
                class="css-j251j9"
                role="cell"
              >
                friedrich-karl.brand@example.com
              </td>
              <td
                class="css-j251j9"
                role="cell"
              >
                Paid
              </td>
            </tr>
            <tr
              class="css-0"
              role="row"
            >
              <td
                class="css-j251j9"
                role="cell"
              >
                Mr. Valentin Ortega
              </td>
              <td
                class="css-j251j9"
                role="cell"
              >
                valentin.ortega@example.com
              </td>
              <td
                class="css-j251j9"
                role="cell"
              >
                Free
              </td>
            </tr>
          </tbody>
        </table>
        <div
          class="css-3mgjd9"
        >
          <button
            class="chakra-button css-1y29bcw"
            data-testid="prev-btn"
            disabled=""
            type="button"
          >
            &lt; Previous
          </button>
          <p
            class="chakra-text css-0"
          >
            Page 
            1
             of 
            5
          </p>
          <button
            class="chakra-button css-1y29bcw"
            data-testid="next-btn"
            type="button"
          >
            Next &gt;
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
